python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -224,27 +222,15 @@ def get_child_partition_keys_of_parent(
             )
 
         partition_mapping = self.get_partition_mapping(child_asset_key, parent_asset_key)
-        downstream_partition_key_range = (
-            partition_mapping.get_downstream_partitions_for_partition_range(
+        downstream_partition_key_subset = (
+            partition_mapping.get_downstream_partitions_for_partition_subset(
                 PartitionKeyRange(parent_partition_key, parent_partition_key),
                 downstream_partitions_def=child_partitions_def,
                 upstream_partitions_def=parent_partitions_def,
             )
         )
 
-        partition_keys = child_partitions_def.get_partition_keys()
```

Comment:
These checks will now occur when attempting to build a partitions subset from a partition key range: https://github.com/dagster-io/dagster/blob/305c6866ac00b23fae4646751eb4c03389af395d/python_modules/dagster/dagster/_core/definitions/partition.py#L228